File: java-src/alchemist-unity/src/main/java/nodes/IMoleculesMap.java
package nodes;

/***
 * This interface defines a structure for a generic class that you can use for serialize and deserialize, for example, a Json message.
 */
public interface IMoleculesMap {

    /***
     * Return the concentration of a given molecule name.
     * @param mol the name of the molecule
     * @return the concentration of the molecule
     */
    Object getMoleculeConcentration(String mol);

    /***
     * Set the internal molecule of this map based on the one passed.
     * @param molecules an instance of another IMoleculeMap you want to copy
     */
    void setMolecules(IMoleculesMap molecules);

    /***
     * Given a molecule name and its concentration, it will set the concentration for the molecule with that name present in this class.
     * @param mol the name of the molecule you want to set
     * @param conc the concentration of the molecule you want to set
     */
    void setMolecule(String mol, Object conc);

}

File: unity3d-src/Examples/SmartCity/Assets/Scripts/Others/IMoleculesMap.cs
/// <summary>
/// 
/// An interface representing the structure of a generic molecule set
/// 
/// </summary>
public interface IMoleculesMap
{
    /// <summary>
    /// 
    /// Method to set a new concentration to a molecule (if present)
    /// 
    /// </summary>
    /// <param name="mol">The molecule to set</param>
    /// <param name="conc">The concentration to set</param>
    void SetMolecule(string mol, object conc);
    /// <summary>
    /// 
    /// Method that allowes to get the concentration of a particular molecule (if present)
    /// 
    /// </summary>
    /// <param name="mol">The molecule to modify</param>
    /// <returns></returns>
    object GetMoleculeConcentration(string mol);
    /// <summary>
    /// 
    /// Set all concentrations with new values
    /// 
    /// </summary>
    /// <param name="molecules">The map with new concentrations</param>
    void SetMolecules(IMoleculesMap molecules);
    /// <summary>
    /// 
    /// Adds a new molecule to the set
    /// 
    /// </summary>
    /// <param name="mol">The name of the molecule</param>
    /// <param name="conc">he concentration</param>
    void AddMolecule(string mol, object conc);
}
